src/main/java/mate/academy/controller/AuthenticationController.java

```diff
@@ -0,0 +1,34 @@
+package mate.academy.controller;
+
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import mate.academy.dto.user.UserLoginRequestDto;
+import mate.academy.dto.user.UserLoginResponseDto;
+import mate.academy.dto.user.UserRegistrationRequest;
+import mate.academy.dto.user.UserResponseDto;
+import mate.academy.exception.RegistrationException;
+import mate.academy.security.AuthenticationService;
+import mate.academy.service.UserService;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/auth")
+public class AuthenticationController {
+    private final UserService userService;
+    private final AuthenticationService authenticationService;
+
+    @PostMapping("/register")
+    public UserResponseDto register(@RequestBody @Valid UserRegistrationRequest request)
+            throws RegistrationException {
+        return userService.register(request);
+    }
+
+    @PostMapping("/login")
+    public UserLoginResponseDto login(@RequestBody UserLoginRequestDto requestDto) {
```

Comment:
You added a field check, but you don't check them
```suggestion
    public UserLoginResponseDto login(@RequestBody @Valid UserLoginRequestDto requestDto) {
```